src/node/share_manager.h

```diff
@@ -266,61 +280,72 @@ namespace ccf
   public:
     ShareManager(NetworkState& network_) : network(network_) {}
 
-    void issue_shares(kv::Tx& tx)
-    {
-      // Assumes that the ledger secrets have not been updated since the
-      // last time shares have been issued (i.e. genesis or re-sharing only)
-      set_recovery_shares_info(
-        tx, network.ledger_secrets->get_latest(tx).second);
-    }
-
-    void issue_shares_on_recovery(kv::Tx& tx, kv::Version latest_ls_version)
+    void issue_recovery_shares(kv::Tx& tx)
     {
+      // Issue new recovery shares for the current ledger secret, recording the
+      // wrapped new ledger secret and encrypted previous ledger secret in the
+      // store.
       auto [latest, penultimate] =
         network.ledger_secrets->get_latest_and_penultimate(tx);
 
-      set_recovery_shares_info(tx, latest, penultimate, latest_ls_version);
+      set_recovery_shares_info(tx, latest.second, penultimate, latest.first);
     }
 
-    void issue_shares_on_rekey(
+    void issue_recovery_shares(
       kv::Tx& tx, const LedgerSecret& new_ledger_secret)
     {
+      // Issue new recovery shares of the new ledger secret, recording the
```

Comment:
Here and below, this would be better as a doxygen block for the function. We don't built an internal API doc (yet?), but if we ever do, we'll be happier that way and it's not much more verbose. It's great to have descriptive comments though, thank you for taking the time to add them!